final_task/rss_reader/rss_parser.py

```diff
@@ -0,0 +1,174 @@
+        output_txt_news method returns result in human readable format (filling the prepared).
+    """
+
+    def __init__(self, args, logger):
+        self.url = args.source
+        self.limit = args.limit
+        self.logger = logger
+
+        self.logger.info(f'Get RSS_url {self.url} and value = {self.limit} (limits amount of output news)')
+
+        self.check_the_connection(self.url, self.logger)
+        self.response = self.get_the_response(self.url, self.logger)
+
+        # Extract the news-site name converting it to the unicode
+        self.feed_name = html.unescape(self.response.feed.get('title', ''))
+        # Extract all news separately in one list
+
+        self.logger.info('Separate a news from the URL.')
+
+        self.news = self.response.entries
+
+        self.limit = self.check_limit_value(self.limit)
+        self.all_news = self.parse_feed()
+
+    @staticmethod
+    def check_the_connection(url, logger):
+        """
+        Check the internet connection
+        :param url: str
+        :param logger: logger
+        :return: None
+        """
+        try:
+            requests.get(url, timeout=1)
+            logger.info('Check the Internet connection.')
+        except (
+                requests.exceptions.ConnectionError,
+                requests.exceptions.Timeout,
+                requests.exceptions.HTTPError,
+        ) as error:
+            raise SystemExit(f'Error: {error}. URL {url} is unreachable.')
+        else:
+            logger.info('Connection established.')
+
+    @staticmethod
+    def check_response_status_code(response, logger):
+        """
+        Check if the response status code is 200: OK
+        :param response: dict
+        :param logger: logger
+        :return: None
+        """
+        if response.get('status', '') != 200:
+            raise Exception(f'Bad response status code {str(response.status)}')
+
+        logger.info('The response status code is 200: OK.')
+
+    def get_the_response(self, url, logger):
+        """
+        Get the response fro the url.
+        :param url: str
+        :param logger: logger
+        :return: dict response
+        """
+        try:
+            response = feedparser.parse(url)
+            self.check_response_status_code(response, logger)
+            logger.info(f'Getting the response from the URL: {url}.')
+        except AttributeError:
+            raise rss_exceptions.InvalidURL("Please, check the URL.")
+        else:
+            logger.info('Valid URL')
+            return response
+
+    def check_limit_value(self, limit):
+        """
+        Check if received limit value is valid
+        :param limit: int: user defined limit value from command line arguments
+        :return: int: valid value of limit variable
+        """
+
+        # total amount of news received from the site
+        total = len(self.news)
+
+        if total == 0:
+            raise rss_exceptions.EmptyLink("Link doesn't contain any news.")
```

Comment:
If you want to check if collection is empty it is better to use:
`if not collection_name:`